src/components/AllSkillsModal.astro

```diff
@@ -0,0 +1,130 @@
+---
+import SkillButton from "./SkillButton.astro";
+
+const API_BASE_URL = import.meta.env.API_BASE_URL;
+const response = await fetch(`${API_BASE_URL}/allSkills`);
+const allSkills = await response.json();
+---
+
+<div>
+  <button
+    type="button"
+    class="relative inline-flex items-center w-full px-4 py-2 text-sm font-medium border-b border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-2 focus:ring-blue-700 focus:text-blue-700 border-gray-600 hover:bg-gray-600 hover:text-white focus:ring-gray-500 focus:text-white"
+    data-drawer-target="drawer2-navigation"
+    data-drawer-show="drawer2-navigation"
+    aria-controls="drawer2-navigation"
+  >
+    <svg
+      class="w-3 h-3 mr-2.5"
+      aria-hidden="true"
+      xmlns="http://www.w3.org/2000/svg"
+      fill="none"
+      viewBox="0 0 20 20"
+    >
+      <path
+        stroke="currentColor"
+        stroke-linecap="round"
+        stroke-linejoin="round"
+        stroke-width="2"
+        d="M7.75 4H19M7.75 4a2.25 2.25 0 0 1-4.5 0m4.5 0a2.25 2.25 0 0 0-4.5 0M1 4h2.25m13.5 6H19m-2.25 0a2.25 2.25 0 0 1-4.5 0m4.5 0a2.25 2.25 0 0 0-4.5 0M1 10h11.25m-4.5 6H19M7.75 16a2.25 2.25 0 0 1-4.5 0m4.5 0a2.25 2.25 0 0 0-4.5 0M1 16h2.25"
+      ></path>
+    </svg>
```

Comment:
Put the source code of the `svg` in a separate component in assets, and then import it here. Makes it more readable and allows Astro to optimise their size.